Sources/Brave/WebFilters/ShieldStats/Adblock/AdBlockStats.swift

```diff
@@ -12,66 +12,228 @@ import os.log
 /// for injected scripts needed during web navigation and cosmetic filters models needed by the `SelectorsPollerScript.js` script.
 public actor AdBlockStats {
```

Comment:
Here the following changes have been made:
1. `resourcesInfo` has been moved to here instead of `FilterListResourcesDownloader`. This is needed for lazy loading.
2. Non-enabled filter lists will be lazily loaded when they are enabled by the user next time the user reloads the page. We store the necessary information that is needed to lazy load in `availableFilterLists`
3. Serial queue is added to synchronously compile engines (one by one) so memory does not climb during critical compilation stages such as launch. Normally only 3 engines are compiled but if the user enables a large amount of filter lists or custom filter lists this, would cause a crash without this serial queue.
4. Handling of memory warnings by deleting non-critical engines. Later perhaps we should disable some filer lists and show a warning to the user.